src/index.js

```diff
@@ -872,7 +881,18 @@ export function useDropzone(props = {}) {
   };
 
   const composeDragHandler = (fn) => {
-    return noDrag ? null : composeHandler(fn);
+    if (noDrag) {
+      return null;
+    }
+    if (ignoreNonFileDragEvents) {
+      return (event, ...args) => {
+        if (!isEvtWithFiles(event)) {
+          return;
+        }
```

Comment:
Here's the heart of it: if the `ignoreNonFileDragEvents` flag is on and the event doesn't have files, we immediately return and let the event do whatever it wants.